### ISSUE TITLE
[clang][nullability] No warning when defererencing _Nullable pointers

### ISSUE THREAD

**Issue by iluuu1994** [llvm/llvm-project#57412]
The [nullability documentation](https://clang.llvm.org/docs/analyzer/developer-docs/nullability.html) states:

> If a pointer p has a nullable annotation and no explicit null check or assert, we should warn in the following cases:
> 
> * p gets implicitly converted into nonnull pointer, for example, we are passing it to a function that takes a nonnull parameter.
> * **p gets dereferenced**

However, I can't get Clang to warn for the latter case.

```c
#include <stdio.h>

int main() {
    int * _Nullable x = NULL;
    printf("%d", *x);

    return 0;
}

```

> -Wnonnull -Wnull-dereference -Wnullability

https://godbolt.org/z/rrGW4h66T

Is there an undocumented flag that needs to be set? Or was this never fully implemented?

**Comment by EugeneZelenko** [comment]
This documentation page is about [Clang Static Analyzer](https://clang.llvm.org/docs/ClangStaticAnalyzer.html) and it's invoked other way.

**Comment by iluuu1994** [comment]
Ah, thank you! Since some warnings work with clang directly (e.g. `-Wnullable-to-nonnull-conversion`) I was under the assumption that they all were.
